### DISCUSSION TITLE
DWM1000 & STM32F103

### DISCUSSION THREAD

**Discussion by hhl-16** [stm32duino/Arduino_Core_STM32#1894] [Q&A] [answered]
hi guys
i use https://github.com/thotro/arduino-dw1000 to stm32 connect dwm1000. it works
I'm running a test sketch basicsender with frequency 3hz ~ T = 300ms
I am using a battery for the circuit, so I want to use it
#include "STM32LowPower.h"
but error
<img width="813" alt="image" src="https://user-images.githubusercontent.com/84115903/206678046-abc6eb83-1257-4969-81c7-21108da99cdc.png">

**Comment by fpistm** [comment, marked as answer]
Hi,

your issue is simply that bot library uses the same name as stated by the logs.
https://github.com/thotro/arduino-dw1000/blob/922ca3e227d7e6aecf841e9b0f9c79eb6036c42f/src/DW1000Constants.h#L27-L30

IDLE_MODE  is a defined value of the dw100 library and in STM32Lowpower it is an enum:
https://github.com/stm32duino/STM32LowPower/blob/082ca1f4f2e8062c81120fd907892e59bffb0151/src/STM32LowPower.h#L53

https://github.com/thotro/arduino-dw1000 is no more maintained (archived), you should try to modify the dw100 library on your side .
